internal/condition/user_condition_test.go

```diff
@@ -0,0 +1,30 @@
+package condition
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-jenkins-idler/internal/testutils/common"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_get_proxy_response(t *testing.T) {
+	// TODO(chmouel): Need to get the actual response from the proxy, built this
+	// json from reading proxy code.
+	tenantData, err := ioutil.ReadFile("../testutils/testdata/proxy.json")
+	if err != nil {
+		assert.NoError(t, err)
+	}
+	srv := common.MockServer(tenantData)
```

Comment:
This is really just testing the happy path. I was hoping to test the error case as well.

Reply:
what do you want me to test more? testing if the json is well passed? this endup testing the tests and just adding tests for sake of adding them, if the statusCode == 200 ? which would me testing httptest library? i am not sure what is the point

Reply:
Maybe I misplaced the comment, but the method under test here is 'uc.getProxyResponse("test")' which is our method. This is the method which creates a request which you respond to in your mock server. As you can see in the function signature, `getProxyResonse` can return a `ProxyResponse` and an error. You are not testing the error case. You just test the happy path with the mock server returning valid JSON.

> testing if the json is well passed? this endup testing the tests and just adding tests for sake of adding them, 

Not at all. The method under test is getProxyResponse. If you look at the implementation, it does low level HTTP calls and JSON marshalling. How are we not testing our production code by not letting the mock server return either invalid or incorrect JSON? 

The case would be a bit different if we were using a Goa client here. There some of the HTTP low level error handling would be covered by the generated code and re-testing it might not be necessary. But in this case adding tests like returning invalid JSON from the mock server or a non 200 HTTP status code would very much improve the test coverage.

> which would me testing httptest library?

No, I wanted to test UserCondition.getProxyResponse. Anyways, let's go without additonal tests then.